apps/backend/src/auth/auth.service.ts

```diff
@@ -0,0 +1,35 @@
+import { AuthSchProfile } from '@kir-dev/passport-authsch';
+import { Injectable } from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { User } from '@prisma/client';
+import { PrismaService } from 'nestjs-prisma';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private prisma: PrismaService,
+    private jwtService: JwtService
+  ) {}
+
+  async findOrCreateUser(userProfile: AuthSchProfile): Promise<User> {
+    const user = await this.prisma.user.findUnique({ where: { authSchId: userProfile.authSchId } });
+    if (user) return user;
+    return await this.prisma.user.create({
+      data: {
+        authSchId: userProfile.authSchId,
+        fullName: `${userProfile.lastName} ${userProfile.firstName}`, //this is kinda cring...
+        nickName: userProfile.firstName,
+        email: userProfile.email,
+        isSchResident: true,
+        //TODO: Find a solution for only dorm residents. The userProfile.bmeStatus is undefined. Is it an admin enabled info?
```

Comment:
And this seems to not work here. Until we find a solution this seems useful for development purposes

Reply:
you need to add the BME_STATUS scope in the constructor to access that field. Still, that won't help with dorm residency.